cmd/container-freezer/main.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"go.uber.org/zap"
+	pkglogging "knative.dev/pkg/logging"
+)
+
+func main() {
+	logger, _ := pkglogging.NewLogger("", "")
+	defer flush(logger)
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		var m ConcurrencyStateMessageBody
+		if r.Body == nil {
```

Comment:
this can't actually happen, request is never nil for a server-side request:

> 
> // For server requests, the Request Body is always non-nil
> // but will return EOF immediately when no body is present.
> // The Server will close the request body. The ServeHTTP
> // Handler does not need to.